File: Message/ArmMessage.py
from Message.Constant import Constant
from Message.Imesssage import Imessage


class ArmMessage(Imessage):
    message_type = None
    arm_type = None

    def __init__(self, arm_type, message_type):
        self.arm_type = arm_type
        self.message_type = message_type

    def get_data(self):
        return self

    def set_data(self, arm_data):
        if arm_data.alert_type == Constant.FACE_RECOGNIZED_ALERT or \
                arm_data.alert_type == Constant.FACE_NOT_RECOGNIZED_ALERT or arm_data.alert_type is None:
            self.arm_type = arm_data.arm_type

    def get_message_type(self):
        return self.message_type

    def set_message_type(self, message_type):
        if message_type == Constant.RESPONSE_MESSAGE or message_type == Constant.AlertMessage or message_type == Constant.PROFILE_UPDATE_MESSAGE or \
                message_type == Constant.CONFIGURE_SETTINGS_MESSAGE or message_type == Constant.Arm_MESSAGE:
            self.message_type = message_type
        else:
            self.message_type = None

File: JavaMessage/ArmMessage.java


public class ArmMessage extends Message implements Imessage{
    
    String armMessage = "ARM_MESSAGE";
    
    String messageType;
    String armType;
    JSONObject messageData;
    
    /**
     * ArmMessage creation from parameter for only armType 
     * message type must be armMessage so assigns into field
     *
     * Creates a JSON object from messageType and armType
     * and stores into messageData
     * 
     * @param armType
     */
    public ArmMessage(String armType){
        this.messageType = armMessage;
        this.armType = armType;
        messageData = new JSONObject();
        messageData.put("messageType", messageType);
        messageData.put("armType", armType);
    }
    
    /**
     * GetData on any message returns the JSON object of the message
     * 
     * @return
     */
    @Override
    public JSONObject getData() {
        return messageData;
    }

    /**
     * Attempt to setData on an ArmMessage object checks that the JSON 
     * contains correct message type (armMessage) before setting the armType
     * from the JSON data.
     * 
     * @param armMsgData
     */
    @Override
    public void setData(JSONObject armMsgData) {
        String temp  = armMsgData.messageType;
        if (temp.equals(armMessage)){
            armType = armMsgData.armType;
        }
    }

    /**
     * Getter
     * 
     * @return
     */
    @Override
    public String getMessageType() {
        return messageType;
    }

    /**
     * Setter should set message type 
     * but only if is correct for this class of message 
     * 
     * @param messageType
     */
    @Override
    public void setMessageType(String messageType) {
        try {
            if(messageType.equals(armType)){
                this.messageType = messageType;
            }
        } catch (Exception e) {
        }
    }
        
    /**
     * Allows for setting armType to zero or one
     * the only permitted values
     * 
     * @param armType
     */
    public void setArmType(String armType){
        try {
            if((armType.equals("0"))||(armType.equals("1"))){
                this.armType = armType;
            }
        } catch (Exception e) {
        }
    }

    /**
     *  Getter
     * 
     * @return
     */
    public String getArmType() {
        return armType;
    }
    
}
